src/nyc_trees/js/src/lib/tabhash.js

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+var $ = require('jquery');
+
+// Tab functions on jquery
+require('bootstrap');
+
+module.exports = function (tabList) {
+    // Use hash to determine which tab to go to
+    var hash = window.location.hash;
+    if (hash) {
+        $(tabList).find('a[href="' + hash + '"]').tab('show');
+    }
+
+    // bootstrap swallows events so the url doesn't change
+    // add our own here
+    $(tabList).on('click', 'a', function (e) {
+        $(this).tab('show');
+
+        var originalScrollPosition = $('body').scrollTop();
+        window.location.hash = this.hash;
+        $('html,body').scrollTop(originalScrollPosition);
+    });
```

Comment:
Rather than intercepting clicks and then triggering a tab to open via `$(this).tab('show')`, consider relying on Bootstrap's events.

Also, I think we can drop the scroll position stuff by using the History API, which [is supported on most recent browsers](http://caniuse.com/#search=replaceState).

This is one option for what this could look like if you follow both of those suggestions:
```javascript
$(tabList).on('show.bs.tab', function(e) {
    var newTab = e.target,
        url = window.location.pathname + '#' + newTab.hash;
    window.history.replaceState(null, null, url);
});